File: src/main/java/commands/slashadapters/adapters/CommandUsagesAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.informationcategory.CommandUsagesCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = CommandUsagesCommand.class)
public class CommandUsagesAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.INTEGER, "page", "Which page to view", false);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(CommandUsagesCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/CommandUsagesAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.informationcategory.CommandUsagesCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = CommandUsagesCommand::class)
class CommandUsagesAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData
            .addOptions(generateOptionData(OptionType.INTEGER, "page", "info_page", false))
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(CommandUsagesCommand::class.java, collectArgs(event))
    }

}